Source/AwsCommonRuntimeKit/mqtt/MqttRequestResponseClient.swift

```diff
@@ -60,17 +51,38 @@ public struct SubscriptionStatusEvent: Sendable {
   public let error: CRTError?
 }
 
-// TODO: Igor has updated the events for IoT Command. Need update later
 /// An event that describes an incoming publish message received on a streaming operation.
 public struct IncomingPublishEvent: Sendable {
 
+  /// The topic associated with this PUBLISH packet.
+  public let topic: String
+
   /// The payload of the publish message in a byte buffer format
-  let payload: Data
+  public let payload: Data
 
-  /// The topic associated with this PUBLISH packet.
-  let topic: String
+  /// (Optional) The content type of the payload
+  public let contentType: String?
+
+  /// (Optional) User Properties, if there is no user property, the array will be empty
+  public let userProperties: [UserProperty]
+
+  /// (Optional) Some services use this field to specify client-side timeouts.
+  public let messageExpiryInterval: TimeInterval?
 
-  // TODO: More options for IoT Command changes
+  /// Internal constructor that setup IncomingPublishEvent from native aws_mqtt_rr_incoming_publish_event
```

Comment:
Trivial: "Internal constructor that initializes an `IncomingPublishEvent` from a native `aws_mqtt_rr_incoming_publish_event`"